software/firmware/europi.py

```diff
@@ -148,7 +149,9 @@ def set_samples(self, samples):
 
     def percent(self, samples=None):
         """Return the percentage of the component's current relative range."""
-        return self._sample_adc(samples) / MAX_UINT16
+        value = self._sample_adc(samples) / MAX_UINT16
+        value = value * (1.0 + 2.0 * self.deadzone) - self.deadzone
+        return max(0.0, min(1.0, value))

```

Comment:
[required] This should use the `clamp()` function that is available in this module.